openmc/filter.py

```diff
@@ -266,35 +215,9 @@ def can_merge(self, other):
 
         """
 
-        if not isinstance(other, Filter):
-            return False
-
-        # Filters must be of the same type
-        if self.type != other.type:
-            return False
-
-        # Distribcell filters cannot have more than one bin
-        if self.type == 'distribcell':
-            return False
+        if type(self) is not type(other): return False
```

Comment:
That wouldn't work because an `energy_filter.can_merge(energy_out_filter)` would return `True`